tests/treactiveSigil.nim

```diff
@@ -147,12 +151,51 @@ suite "reactive examples":
     let z = computed[float]():
       x{} * y{}
 
-    echo "X: ", x.val, " Z: ", z.val
-    check isNear(x.val, 3.14)
-    check isNear(y.val, 2.718)
-    check isNear(z.val, 8.53451979637.float, 4)
+    echo "X: ", x{}, " Z: ", z{}
+    check isNear(x{}, 3.14)
+    check isNear(y{}, 2.718)
+    check isNear(z{}, 8.53451979637.float, 4)
 
     x <- 1.0
-    check isNear(x.val, 1.0)
-    check isNear(y.val, 2.718)
-    check isNear(z.val, 2.718)
+    check isNear(x{}, 1.0)
+    check isNear(y{}, 2.718)
+    check isNear(z{}, 2.718)
+
+  test "reactive wrapper and lazy compute":
+
+    var cnt: ref int
+    cnt.new()
+
+    let
+      x = newSigil(5)
+      z = computedLazy[int]():
+        cnt[].inc()
+        8 * x{}
+
+    when defined(sigilsDebug):
+      x.debugName = "X"
+      z.debugName = "Z"
+
+    check cnt[] == 0 # cnt is called from the `read` (`trace`) setup step
+    check cnt[] == 0
+    x <- 2
+    echo "X: ", x{},  " => Z: ", z{}, " (", cnt[], ")"
+    check z{} == 16
+    check cnt[] == 1
+    x <- 2
+    echo "X: ", x{},  " => Z: ", z{}, " (", cnt[], ")"
+    check cnt[] == 1
+    check z{} == 16
+    echo "X: ", x{},  " => Z: ", z{}, " (", cnt[], ")"
+    check cnt[] == 1
+    ## shouldn't trigger changes
+    x <- 2
+    check cnt[] == 1
+    check z{} == 16
+    check cnt[] == 1
+
+    ## shouldt trigger changes after read
```

Comment:
This should be "should trigger change after read"

Reply:
> This should be "should trigger change after read"

Well I wasn't sure so I left it in between "should" and "shouldn't". Totally on purpose and not a typo :P